src/cursor/cur_layered.c

```diff
@@ -640,33 +644,30 @@ __clayered_get_current(
         else
             current = clayered->stable_cursor;
 
-        /*
-         * If the cursors are equal, choose the ingest cursor to return the result but remember not
-         * to later return the same result from the stable cursor.
-         */
-        if (cmp == 0)
```

Comment:
I get that `WT_CLAYERED_MULTIPLE` was never read, but I wonder if the removal of this code points to some other bug -- do you know what is meant to happen in the case the old comment describes, where the cursors are equal?

Reply:
TBH I don't see logic behind this comment. Let's imagine a situation where we have two constituents that have overalps:
```
ingest: [     3 4 5      8     ]
stable: [ 1 2 3        7 8 9 ] 
```

When we iterate to the first element that exists in both tables we would have both ingest and stable set to 3 and `current_cursor` would be set to ingest. On the next step as a part of `next()` or `prev()` we will identify `alternate_cursor` (stable in this case) and if it has the same value with current_cursor, it will call `prev()` or `next()` for it as well.

Please see this logic [here](https://github.com/wiredtiger/wiredtiger/pull/12425/files#diff-4cb84d228283c84bb09fb834c30439d13032f06a709d9dc74c90354e293f12efR835).

Reply:
Maybe we can avoid calling extra `__clayered_cursor_compare` and use `WT_CLAYERED_MULTIPLE` instead, but it requires more time to investigate to cover all the cases. If you think that it could be useful improvement, I can create a ticket and add a FIXME for now

Reply:
Thanks for the explanation, it helps a lot. I think I understand what you're proposing, and it sounds like an improvement. But for performance stuff, I always prefer to let profiling find the issues, so I don't feel strongly about making a ticket.